packages/atomic/src/components/search/result-template-components/quickview-word-highlight/quickview-word-highlight.ts

```diff
@@ -0,0 +1,68 @@
+import {ColorParser} from './color-parser';
+
+export {getWordsHighlights, HIGHLIGHT_PREFIX} from './iframe-parser';
+
+export class QuickviewWordHighlight {
+  public readonly focusedColor: string;
+  public readonly previewBorderColor: string;
+  public readonly elements: HTMLElement[] = [];
+
+  private currentNavigationPosition = -1;
+
+  constructor(
+    public readonly indexIdentifier: string,
+    public readonly text: string,
+    public readonly color: string,
+    element: HTMLElement
+  ) {
+    this.addElement(element);
+
+    const colorParser = new ColorParser(color);
+    this.focusedColor = colorParser.rgbInverted();
+    this.previewBorderColor = colorParser.rgbSaturated();
+  }
+
+  public addElement(element: HTMLElement) {
+    this.elements.push(element);
+  }
+
+  public get occurrences() {
+    return this.elements.length;
+  }
+
+  public navigateForward() {
+    this.currentNavigationPosition++;
+    if (this.currentNavigationPosition >= this.elements.length) {
+      this.currentNavigationPosition = 0;
+    }
+    this.highlightNavigation();
+    this.putElementIntoView();
+    return this.elements[this.currentNavigationPosition];
+  }
+
+  public navigateBackward() {
+    this.currentNavigationPosition--;
+    if (this.currentNavigationPosition < 0) {
+      this.currentNavigationPosition = this.elements.length - 1;
+    }
+    this.highlightNavigation();
+    this.putElementIntoView();
+    return this.elements[this.currentNavigationPosition];
+  }
+
+  private highlightNavigation() {
+    const currentElement = this.elements[this.currentNavigationPosition];
+    const otherElements = this.elements.filter((el) => el !== currentElement);
```

Comment:
This is an expensive way to filter out an element whose position we know.
This, combined with the forEach l58, means we iterate twice on this.elements (minus one element that we skip).

Unfortunately, Arrays in JS are akin to a stack, so removing items anywhere but at the end is "expensive".

Here, I would convert the forEach in for..of and use a guard (`if(otherElement===currentElement) continue;`), it would increase _slightly_ the cost per iteration with the 2 reads + comparaison, but still be cheaper than the current approach